utils/build/docker/golang/parametric/datadog.go

```diff
@@ -101,6 +107,80 @@ func (s *apmClientServer) SpanSetError(ctx context.Context, args *SpanSetErrorAr
 	return &SpanSetErrorReturn{}, nil
 }
 
+type CustomLogger struct {
+	*logrus.Logger
+	globalConfig map[string]string
+}
+
+type Config struct {
+	Service                string            `json:"service"`
+	SampleRate             string            `json:"sample_rate"`
+	RuntimeMetricsEnabled  bool              `json:"runtime_metrics_enabled"`
+	Tags                   map[string]string `json:"tags"`
+	PropagationStyleInject string            `json:"propagation_style_inject"`
+	Debug                  bool              `json:"debug"`
+	Env                    string            `json:"env"`
+	DdVersion              string            `json:"dd_version"`
+}
+
+func (l *CustomLogger) Log(logMessage string) {
+	re := regexp.MustCompile(`DATADOG TRACER CONFIGURATION (\{.*\})`)
+	matches := re.FindStringSubmatch(logMessage)
+	if len(matches) < 2 {
+		log.Print("JSON not found in log message")
+		return
+	}
+	jsonStr := matches[1]
+
+	var config Config
+	if err := json.Unmarshal([]byte(strings.ToLower(jsonStr)), &config); err != nil {
+		log.Print("Error unmarshaling JSON: %v\n", err)
+		return
+	}
+
+	stringConfig := make(map[string]string)
+	val := reflect.ValueOf(config)
+	for i := 0; i < val.Type().NumField(); i++ {
+		field := val.Type().Field(i)
+		valueField := val.Field(i)
```

Comment:
Okay, so you're essentially doing all this logic with reflect in order to iterate over the struct fields and values. Maybe add a comment above line 142 prefacing where you're going, since it wasn't immediately clear to me.